### PR TITLE
feat:added the working of all the buttons in PWM panel

### LINKED ISSUE #1009: Implement the working of the PWM panel
**Actual Behaviour**

The PWM panel in wave generator activity is not working.

**Expected Behaviour**

All the buttons in PWM panel sq1, sq2, sq3, sq4, frequency, duty, phase should be working and connected to the PWM monitor.

**Steps to reproduce it**

WaveGenerator Activity->Implement working of PWM panel.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Avjeet**
Fixes #1009 
Partly fixes the issue #878 

**Changes**:

- All the buttons in the PWM panel are working.
- Complete UI of wave generator is working.

This PR requires #973 and #983 to be merged first.

**Screenshot/s for the changes**: 
ezgif com-video-to-gif 2

**Checklist**: 
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

> @Avjeet -- Note: Don't blindly check these boxes. It's a self check so please check only if the task is done. I've unchecked the boxes with incomplete tasks

**APK for testing**: [app-debug.zip](https://github.com/fossasia/pslab-android/files/2087256/app-debug.zip)

### CODE REVIEW

**Comment by Avjeet** [comment]
@CloudyPadmal @abhinavraj23 @harsh-2711 @Vikum94 @akarshan96 Plz reviews the working of the complete UI on your devices.
When the previous PR #973 and #983 are merged then all the previous commits will be removed.

File: app/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -214,4 +215,5 @@
     <dimen name="connect_instruction_text_header">18sp</dimen>
 
     <dimen name="sensor_schematic_height">250dp</dimen>
+
```
Comment (CloudyPadmal) [inline app/src/main/res/values/dimens.xml]: You have checked the item in PR checklist stating that no modifications done at the end of resource files. But here there is a change making merge conflicts.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -746,4 +756,4 @@
     <string name="ir_thermometer_desc">\u2022 The above pin configuration has to be same except for the pin
     GND. You can use any of the PSLab device GND pins.\n</string>
 
-</resources>
+</resources>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: Here also. @Avjeet , @abhinavraj23 , @harsh-2711 , @Vikum94 Guys please don't remove this system generated new line character at each file explicitly.

Comment (Vikum94) [reply app/src/main/res/values/strings.xml]: Noted

**Comment by Vikum94** [comment]
Code looks good. Is there a specific reason why you have not squashed the commits?

File: app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java [hunk 3]
```diff
@@ -197,44 +341,137 @@ public void onStopTrackingTouch(IndicatorSeekBar seekBar) {
...
-        wavePhaseValue.setText(valueText);
+                value = (double) WaveGeneratorCommon.wave.get(pwmBtnActive).get(DUTY);
+                valueText = value + " " + getString(R.string.unit_percent);
+                pwmDutyValue.setText(valueText);
+
+                value = (double) WaveGeneratorCommon.wave.get(pwmBtnActive).get(PHASE);
+                valueText = value + " " + getString(R.string.deg_text);
+                pwmPhaseValue.setText(valueText);
+                break;
+        }
```
Comment (Vikum94) [inline app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java]: add the default label for this switch block

File: app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java [hunk 4]
```diff
@@ -68,17 +107,34 @@
...
     //const values
+    public static final int FREQ_MIN = 10;
+    public static final int DUTY_MIN = 0;
+    public static final int PHASE_MIN = 0;
+
+    public static final int FREQ_MAX = 5000;
+    public static final int DUTY_MAX = 100;
+    public static final int PHASE_MAX = 360;
+
     public static final int SIN = 1;
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java]: You can easily use Java `enum` for these kind of variables.

Comment (Avjeet) [reply app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java]: If there is no other change could u plz merge it, I will address this changes in another PR as it is a lengthy task.

Comment (Avjeet) [reply app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java]: If there are no other changes could u plz merge this PR as this enum task is lengthy and I will address that in another PR,

Comment (CloudyPadmal) [reply app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java]: Then can you open an issues and link it here so we have a record of you doing it in future?

**Comment by Avjeet** [comment]
@CloudyPadmal @abhinavraj23 changes done

**Comment by Avjeet** [comment]
@CloudyPadmal plz merge this

**Comment by CloudyPadmal** [comment]
@Avjeet other developers need to review this before we can merge it (Best practices mention this to be two or more). Please don't rush it and wait for others to review.

**Pull request merged**